src/users/amoc_users_sup.erl

```diff
@@ -120,47 +127,53 @@ start_children(Scenario, UserIds, ScenarioState) ->
 %% in order to load-balance the request among all workers.
 -spec stop_children(non_neg_integer(), boolean()) -> non_neg_integer().
 stop_children(Count, Force) ->
-    {CountRemove, Assignments} = assign_counts(Count),
-    [ amoc_users_worker_sup:stop_children(Sup, Int, Force) || {Sup, Int} <- Assignments ],
-    CountRemove.
+    Users = case ets:match_object(?TABLE, '$1', Count) of
+               '$end_of_table' ->
+                   [];
+               {Objects, _} ->
+                   Objects
+           end,
+    stop_children_assignments(Users, Force),
+    length(Users).
 
 -spec get_all_children() -> [{pid(), amoc_scenario:user_id()}].
 get_all_children() ->
```

Comment:
seems that this function is used for testing only, we can remove it and rework testing a bit.

Reply:
Not sure how without literally doing ets:tab2list from tests instead of here, but that'd make this one a leaky abstraction 🤔